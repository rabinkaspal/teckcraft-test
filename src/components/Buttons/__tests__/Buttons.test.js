import { fireEvent, render, screen } from "@testing-library/react";
import CatFactsButton from "..";
import renderer from "react-test-renderer";

const mockedBtnOnClick = jest.fn();

describe("Button", () => {
  it("should have same label as text props", () => {
    render(<CatFactsButton text="Cat Fact Button Test" />);
    const btnElement = screen.getByText("Cat Fact Button Test");
    expect(btnElement).toBeInTheDocument();
  });

  it("should show loading text when props loading true", () => {
    render(<CatFactsButton text="Get Cat Fact" loading={true} />);
    const btnElement = screen.getByText("Loading...");
    expect(btnElement).toBeInTheDocument();
  });

  it("should call btnOnClick on click", () => {
    render(<CatFactsButton text="Cat Fact Button" btnOnClick={mockedBtnOnClick} />);
    let btnElement = screen.getByText("Cat Fact Button");
    expect(btnElement).toBeInTheDocument();

    fireEvent.click(btnElement);
    expect(mockedBtnOnClick).toBeCalled();
  });

  it("should be disabled when loading props is true", () => {
    render(<CatFactsButton text="Cat Fact Button" loading={true} />);
    let btnElement = screen.getByText("Loading...");
    expect(btnElement).toBeDisabled();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<CatFactsButton text="Get a new fact" />);
    expect(tree).toMatchSnapshot();
  });
});
