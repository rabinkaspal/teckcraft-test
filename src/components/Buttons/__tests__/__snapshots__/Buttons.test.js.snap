// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="sc-gswNZR bPbNkq"
  data-testid="fact-button"
>
  Get a new fact
</button>
`;
